src/libmain/progress-bar.cc

```diff
@@ -109,7 +109,12 @@ class ProgressBar : public Logger
     {
         {
             auto state(state_.lock());
-            if (!state->active) return;
+            if (!state->active) {
+                // wait for the thread to complete to avoid
+                // 'terminate called without an active exception'
+                if (updateThread.joinable()) updateThread.join();
+                return;
+            }
```

Comment:
Would love feedback here.
This was kind of tricky because it looked like in some cases the ProgressBar class was destroyed while the thread was still active which raised an error.

I then tried to add `join` but I believe that there is a timing issue here that the thread was not even started.
Still feels like a data race.